src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -61,22 +61,40 @@ public void storeMessage(Message message) {
             .addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
-    for (Entity entity : results.asIterable()) {
+    createMessage(results, messages);
+
+    return messages;
+  }
+
+  public List<Message> getAllMessages() {
+    List<Message> messages = new ArrayList<>();
+
+    Query query = new Query("Message")
+      .addSort("timestamp",SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+
+    createMessage(results, messages);
+
+    return messages;
+  }
+
+  private void createMessage(PreparedQuery pq, List<Message> m) {
```

Comment:
Nice job extracting the shared logic, but rather than taking a List as a parameter and populating it within createMessage(), why not just construct the List within createMessage() and return it? This also cleans up the places where you call createMessage() so you can just do something like `return createMessage(results)`.